protocols/kad/src/behaviour.rs

```diff
@@ -283,6 +315,18 @@ impl<TSubstream> Kademlia<TSubstream> {
         self.start_query(QueryInfoInner::GetProviders { target, pending_results: Vec::new() });
     }
 
+    /// Starts an iterative `GET_VALUE` request.
+    pub fn get_data(&mut self, key: Multihash) {
+        self.start_query(QueryInfoInner::GetValue { key, results: vec![] });
+    }
+
+    /// Starts an iterative `PUT_VALUE` request
+    pub fn put_data(&mut self, key: Multihash, data: &[u8]) {
+        // TODO: Probably we shouldn't store the value ourselves
+        self.records.insert(key.clone(), data.to_vec());
```

Comment:
Is it intentional that the record is inserted here and again later when the `PutValue` query completes [here](https://github.com/libp2p/rust-libp2p/pull/1144/files#diff-88d683446d7fa7abbd7a347fb7d7c540R850)?

Reply:
Here we put the data on the side of the putter